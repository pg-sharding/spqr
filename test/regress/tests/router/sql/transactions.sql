\c spqr-console
CREATE DISTRIBUTION ds1 COLUMN TYPES integer;
CREATE KEY RANGE krid1 FROM 1 ROUTE TO sh1 FOR DISTRIBUTION ds1;
CREATE KEY RANGE krid2 FROM 101 ROUTE TO sh2 FOR DISTRIBUTION ds1;
ALTER DISTRIBUTION ds1 ATTACH RELATION transactions_test DISTRIBUTION KEY id;

\c regress

CREATE TABLE transactions_test (id int);

-- check that rollbacked changes do no apply
BEGIN;
SELECT * FROM transactions_test WHERE id = 1;;
INSERT INTO transactions_test (id) VALUES (1);
SELECT * FROM transactions_test WHERE id = 1;;
ROLLBACK;

SELECT * FROM transactions_test WHERE id = 1;;

-- check that commited changes present
BEGIN;
SELECT * FROM transactions_test WHERE id = 1;;
INSERT INTO transactions_test (id) VALUES (1);
SELECT * FROM transactions_test WHERE id = 1;;
COMMIT;

SELECT * FROM transactions_test WHERE id = 1;;

DROP TABLE transactions_test;

\c spqr-console
DROP DISTRIBUTION ALL CASCADE;
DROP KEY RANGE ALL;
