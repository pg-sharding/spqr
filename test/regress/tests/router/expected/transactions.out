\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

CREATE DISTRIBUTION ds1 COLUMN TYPES integer;
         add distribution         
----------------------------------
 created distribution with id ds1
(1 row)

CREATE KEY RANGE krid1 FROM 1 ROUTE TO sh1 FOR DISTRIBUTION ds1;
         add key range          
--------------------------------
 created key range with bound 1
(1 row)

CREATE KEY RANGE krid2 FROM 101 ROUTE TO sh2 FOR DISTRIBUTION ds1;
          add key range           
----------------------------------
 created key range with bound 101
(1 row)

ALTER DISTRIBUTION ds1 ATTACH RELATION transactions_test DISTRIBUTION KEY id;
                      attach table                       
---------------------------------------------------------
 attached relation transactions_test to distribution ds1
(1 row)

\c regress
CREATE TABLE transactions_test (id int);
NOTICE: send query to shard(s) : sh1,sh2
-- check that rollbacked changes do no apply
BEGIN;
SELECT * FROM transactions_test WHERE id = 1;;
NOTICE: send query to shard(s) : sh1
 id 
----
(0 rows)

INSERT INTO transactions_test (id) VALUES (1);
SELECT * FROM transactions_test WHERE id = 1;;
 id 
----
  1
(1 row)

ROLLBACK;
SELECT * FROM transactions_test WHERE id = 1;;
NOTICE: send query to shard(s) : sh1
 id 
----
(0 rows)

-- check that commited changes present
BEGIN;
SELECT * FROM transactions_test WHERE id = 1;;
NOTICE: send query to shard(s) : sh1
 id 
----
(0 rows)

INSERT INTO transactions_test (id) VALUES (1);
SELECT * FROM transactions_test WHERE id = 1;;
 id 
----
  1
(1 row)

COMMIT;
SELECT * FROM transactions_test WHERE id = 1;;
NOTICE: send query to shard(s) : sh1
 id 
----
  1
(1 row)

DROP TABLE transactions_test;
NOTICE: send query to shard(s) : sh1,sh2
\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

DROP DISTRIBUTION ALL CASCADE;
   drop distribution   
-----------------------
 drop distribution ds1
(1 row)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

